src/Coalesce.Web/tsconfig.json

```diff
@@ -1,10 +1,6 @@
 {
   "compileOnSave": false,
   "compilerOptions": {
-    "strict": true,
```

Comment:
Our strictness settings here are desired - we want to make sure that the typescript we generate can work with as many strict settings as possible (unfortunately, we can't manage to achieve `strictFunctionTypes`, and `strictPropertyInitialization` was off because VS would blow up using the definite assignment operator on properties in VS versions that don't support TS 2.7 (although this shouldn't be an issue anymore).

Its OK that we don't have these in the starter project - I wouldn't impose `strictNullChecks` on even my worst enemies without their explicit consent.

Reply:
That's going to require some hacking with the generators or base classes then, because TS 2.8.1 throws a bunch of `error TS2344: Type 'xyz' does not satisfy the constraint 'ClientMethodParent'.` This happened in 2.4 when they added the weak type checkers: https://blogs.msdn.microsoft.com/typescript/2017/06/12/announcing-typescript-2-4-rc/
I can turn off 
```
    "strictFunctionTypes": false,
    "strictPropertyInitialization": false 
```
To get it to compile without errors.

Reply:
Ok, I'll take a look

Reply:
Oh right, that's what I meant - unfortunately, the comments on github only shows context above the commented line and not below. We do need those two off.

Reply:
I may need to pair with you to understand what tsc thinks is wrong with obviously working TypeScript code.  I hope a small change to the base classes or the generated code will allow the strict-mode exceptions set above can be turned off.